### PR TITLE
feat(UNT-T10098): implement animated wallet component

### PR DESCRIPTION
**Pull request by meet-p-simform**
-Implemented Animated Wallet Component with some required  props.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -105,5 +106,8 @@
     "extends": [
       "@commitlint/config-conventional"
     ]
+  },
+  "dependencies": {
+    "chroma-js": "^2.4.2"
```
Comment (shalin-simform) [inline package.json:111]: move this into dev-dependencies

File: src/components/AnimatedWallet/AnimatedWallet.tsx [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+import { View } from 'react-native';
+import LinearGradient from 'react-native-linear-gradient';
+import { useAnimatedWallet } from './hook';
+import styles from './styles';
+import type { AnimatedWalletStyle } from './types';
+
+const AnimatedWallet = ({
+  size = 250,
+  primaryColor = '#f51e0f',
+  secondaryColor = '#f07a05',
```
Comment (shalin-simform) [inline src/components/AnimatedWallet/AnimatedWallet.tsx:11]: import color from Colors file

File: src/components/AnimatedWallet/hook/useAnimatedWallet.tsx [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+
+const useAnimatedWallet = ({
+  size,
+  primaryColor = '',
+  secondaryColor = '',
+}: AnimatedWalletStyle) => {
+  const AnimatedGradientView = Animated.createAnimatedComponent(LinearGradient);
+  const animation = useRef(new Animated.Value(0)).current;
+
+  const lighterPrimaryColor = chroma(primaryColor).brighten().hex();
```
Comment (shalin-simform) [inline src/components/AnimatedWallet/hook/useAnimatedWallet.tsx:15]: add optional chaining

File: package.json (hunk 1 above)
Comment (meet-p-simform) [reply package.json:111]: Resolved

File: src/components/AnimatedWallet/AnimatedWallet.tsx (hunk 2 above)
Comment (meet-p-simform) [reply src/components/AnimatedWallet/AnimatedWallet.tsx:11]: Resolved

File: src/components/AnimatedWallet/hook/useAnimatedWallet.tsx (hunk 3 above)
Comment (meet-p-simform) [reply src/components/AnimatedWallet/hook/useAnimatedWallet.tsx:15]: Resolved

File: src/components/AnimatedWallet/hook/useAnimatedWallet.tsx [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+        rotateZ: animation.interpolate({
+          inputRange: [0, 1, 2],
+          outputRange: ['0deg', '-15deg', '0deg'],
+          extrapolateRight: 'clamp',
+        }),
+      },
+      {
+        translateY: animation.interpolate({
+          inputRange: [0, 1, 2],
+          outputRange: [0, -size / 5, 0],
```
[the remark this answers is not in the document]
Comment (meet-p-simform) [reply src/components/AnimatedWallet/hook/useAnimatedWallet.tsx:45]: Resolved (It need -size).

File: src/components/AnimatedWallet/AnimatedWallet.tsx [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+          {...gradientPosition}
+        >
+          <AnimatedGradientView
+            colors={[lighterPrimaryColor, primaryColor]}
+            style={[style.circle]}
+            {...gradientPosition}
+          />
+        </AnimatedGradientView>
+      </AnimatedGradientView>
+      {(numberOfCards === 1 || numberOfCards === 2 || numberOfCards === 3) && (
```
Comment (shalin-simform) [inline src/components/AnimatedWallet/AnimatedWallet.tsx:54]: optimise this condition

File: src/components/AnimatedWallet/AnimatedWallet.tsx [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+        <AnimatedGradientView
+          colors={[primaryColor, lighterPrimaryColor, primaryColor]}
+          style={[{ ...style.card }, style.card1, card1Animation]}
```
Comment (shalin-simform) [inline src/components/AnimatedWallet/AnimatedWallet.tsx:57]: update the name of style and animation variable

File: src/components/AnimatedWallet/AnimatedWallet.tsx (hunk 5 above)
Comment (meet-p-simform) [reply src/components/AnimatedWallet/AnimatedWallet.tsx]: Resolvd

File: src/components/AnimatedWallet/AnimatedWallet.tsx (hunk 6 above)
Comment (meet-p-simform) [reply src/components/AnimatedWallet/AnimatedWallet.tsx]: Resolved

**Review by shalin-simform — APPROVED** [review]

**Pull request merged**